File: java-martian-ddd/src/Grid.java
final class Grid { final int maxX, maxY; Grid(int maxX,int maxY){ this.maxX=maxX; this.maxY=maxY; }
  boolean contains(Position p){ return 0<=p.x && p.x<=maxX && 0<=p.y && p.y<=maxY; }
}

File: csharp-martian-ddd/src/MartianDDD.Domain/Grid.cs
namespace MartianDDD.Domain;

public sealed class Grid
{
    public int MaxX { get; }
    public int MaxY { get; }

    public Grid(int maxX, int maxY)
    {
        if (maxX < 0 || maxY < 0) throw new ArgumentOutOfRangeException("Grid dimensions must be non-negative");
        MaxX = maxX;
        MaxY = maxY;
    }

    public bool Contains(Position p) => p.X >= 0 && p.X <= MaxX && p.Y >= 0 && p.Y <= MaxY;
}
